include/nbl/asset/metadata/IRenderpassIndependentPipelineMetadata.h

```diff
@@ -103,16 +104,15 @@ class IRenderpassIndependentPipelineMetadata : public core::Interface
 
 				auto operator<=>(const PushConstant&) const = default;
 			};
-
-			enum E_TYPE
+			enum class E_TYPE: uint8_t 
 			{
-				ET_COMBINED_IMAGE_SAMPLER = IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER,
-				ET_STORAGE_IMAGE = IDescriptor::E_TYPE::ET_STORAGE_IMAGE,
-				ET_UNIFORM_TEXEL_BUFFER = IDescriptor::E_TYPE::ET_UNIFORM_TEXEL_BUFFER,
-				ET_STORAGE_TEXEL_BUFFER = IDescriptor::E_TYPE::ET_STORAGE_TEXEL_BUFFER,
-				ET_UNIFORM_BUFFER = IDescriptor::E_TYPE::ET_UNIFORM_BUFFER,
-				ET_STORAGE_BUFFER = IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
-				ET_INPUT_ATTACHMENT = IDescriptor::E_TYPE::ET_INPUT_ATTACHMENT,
+				ET_COMBINED_IMAGE_SAMPLER = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER),
+				ET_STORAGE_IMAGE = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_STORAGE_IMAGE),
+				ET_UNIFORM_TEXEL_BUFFER = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_UNIFORM_TEXEL_BUFFER),
+				ET_STORAGE_TEXEL_BUFFER = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_STORAGE_TEXEL_BUFFER),
+				ET_UNIFORM_BUFFER = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_UNIFORM_BUFFER),
+				ET_STORAGE_BUFFER = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_STORAGE_BUFFER),
+				ET_INPUT_ATTACHMENT = static_cast<uint8_t>(IDescriptor::E_TYPE::ET_INPUT_ATTACHMENT),
```

Comment:
Would you mind making a little utility in `nbl/core/util/to_underlying.h`
```cpp
namespace nbl::core
{
#if defined(__cpp_lib_to_underlying)
using to_underlying = std::to_underlying;
#else
template<typename E>
constexpr auto to_underlying(E e) -> typename std::underlying_type<E>::type 
{
   return static_cast<typename std::underlying_type<E>::type>(e);
}
#endif
}
```

or something equivalent, since I have no clue if `using` can be used to alias a template function